package.json

```diff
@@ -375,6 +375,11 @@
             "default": {},
             "description": "Specifies any environment variables your project needs. (Probably mostly for your Workspace Settings.)"
           },
+          "calva.jackInProjectType": {
+            "type": "string",
+            "default": null,
+            "description": "Specifies the project type for Jack-in by default. (Probably mostly for your Workspace Settings.)"
```

Comment:
> (Probably mostly for your Workspace Settings.)

Can you elaborate on what this means?

Reply:
I think we should specify that this allows bypassing the prompt for the project type at jack-in. Maybe we can say "Specifies the project type for jack-in, which causes the project type menu at jack-in to be bypassed."

Reply:
@PEZ Do we need to list the possible values here somehow?

@SillyCoon Maybe we can use a dropdown and just list all the possible project types. It provides a more specific schema for the setting. In either case, they can put something that won't work,  but at least with a dropdown the setting isn't as susceptible to a spelling mistake, even if they edit it in the json and not via the settings UI (VS Code will suggest the possible string values if it's a dropdown, I think).

Reply:
Thank you for the comment! 
- (Probably mostly for your Workspace Settings.) - I copied it from calva.jackInEnv. In my understanding, it means that this option is useful only in your Workspace `settings.json`, and not as a setting for the entire VS Code
- I picked up the naming from here https://github.com/BetterThanTomorrow/calva/pull/2073#issuecomment-1425411118
- In this case, I'm afraid that dropdown doesn't support custom sequences defined by user

Reply:
> I picked up the naming from here https://github.com/BetterThanTomorrow/calva/pull/2073#issuecomment-1425411118

That works.

> In this case, I'm afraid that dropdown doesn't support custom sequences defined by user

What do you mean here? I think the dropdown can just list all project types that Calva supports.

Reply:
I'd like @PEZ's opinion here though. If we add a new project type later we'd have to be sure to add to this dropdown, but if the dropdown pulls from the same list of project types that are used elsewhere, then adding a new one would automatically add the option to this dropdown. I don't know if the options are centralized like that at the moment, though.

Reply:
I'm not sure what is discussed here. I have tried the PR and custom sequences seem to work just fine.

Reply:
The user has to type out the string themselves for the project type. Typos would lead to an error, right? That's probably not a big deal though. I think we should at least list out the possible values for this setting in the docs.

Reply:
And of course some might not apply to certain projects, but we should state that as well.

Reply:
The possible values are not known, it depends on the project and what custom REPL sequences and stuff. We can mention in the description that the value to fill in will be printed in the Output/REPL window when they connect to the REPL.

Reply:
Yeah, that's the main problem, we don't know all the possible projects to suggest until the runtime, and we cannot change the dropdown for the settings on runtime too, it seems to be one of the limitations of the VS code itself 😅 So the only simple way to be sure that we're putting the correct project to the setting is to use the `jack-in` dropdown, and to notify the user in the Output Window that the setting is incorrect if it was specified with a typo.

Reply:
Oh, I forgot about custom repl sequences being in that list.

> We can mention in the description that the value to fill in will be printed in the Output/REPL window when they connect to the REPL.

> So the only simple way to be sure that we're putting the correct project to the setting is to use the jack-in dropdown, and to notify the user in the Output Window that the setting is incorrect if it was specified with a typo.

This sounds good. I'm wondering if the output window is the best place to inform them of an issue with the setting though, since the output window may not be open at that time. I think in most cases it will open when they start to jack-in, but some people may have the setting to not open the output window at jack-in enabled. Maybe _this_ is a place to use a pop-up.

Reply:
I just saw this comment from @PEZ:

> For the last case there I think we should show the menu, which will then make us print the correct settings value.

That makes sense to me.